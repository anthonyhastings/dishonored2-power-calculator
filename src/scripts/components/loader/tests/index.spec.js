import React from 'react';
import {render} from 'enzyme';
import Loader from '../';

const renderLoader = function (props) {
  return (
    <Loader {...props}>
      <p>Hello World</p>
    </Loader>
  );
};

describe('Loader component', function () {
  describe('when loading is incomplete', function () {
    beforeEach(function () {
      const loader = renderLoader();

      this.wrapper = render(loader);
    });

    it('renders the loaders default content', function () {
      expect(this.wrapper).toMatchSnapshot();
    });
  });

  describe('when loading is complete', function () {
    beforeEach(function () {
      const loader = renderLoader({loaded: true});

      this.wrapper = render(loader);
    });

    it('renders the loaders children', function () {
      expect(this.wrapper).toMatchSnapshot();
    });
  });

  describe('when loading has errored', function () {
    beforeEach(function () {
      const loader = renderLoader({loaded: new Error()});

      this.wrapper = render(loader);
    });

    it('renders the loaders error children', function () {
      expect(this.wrapper).toMatchSnapshot();
    });
  });
});
