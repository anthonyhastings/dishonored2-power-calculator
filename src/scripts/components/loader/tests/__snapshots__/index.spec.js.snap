// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component when loading has errored renders the loaders error children 1`] = `
<div
  class="app-loader"
>
  <h2
    class="app-loader__message"
  >
    Error while loading
  </h2>
  <div
    class="app-loader__icon-container"
  />
</div>
`;

exports[`Loader component when loading is complete renders the loaders children 1`] = `
<p>
  Hello World
</p>
`;

exports[`Loader component when loading is incomplete renders the loaders default content 1`] = `
<div
  class="app-loader app-loader--is-loading"
>
  <h2
    class="app-loader__message"
  >
    Loading
  </h2>
  <div
    class="app-loader__icon-container"
  />
</div>
`;
